### PR TITLE
internal/gocdk: add a docstore demo

### LINKED ISSUE #2354: internal/gocdk: add a demo for docstore

### PR DESCRIPTION
**Pull request by vangent**
Fixes #2354.

### CODE REVIEW

File: internal/cmd/gocdk/internal/static/_assets/init/go.mod [hunk 1]
```diff
@@ -2,4 +2,4 @@ module {{.ModulePath}}
 
 go 1.12
 
-require gocloud.dev v0.15.0
+require gocloud.dev v0.15.1-0.20190626205612-1012b9d3127e
```
Comment (vangent) [inline internal/cmd/gocdk/internal/static/_assets/init/go.mod]: Note: I had to set this to a pseudo-version in order to load docstore. As soon as we have a real release with docstore we should update this.

File: internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+type MyDocument struct {
+	Key   string
+	Value string
+	// TODO(rvangent): Rename this field to demonstrate setting it once
+	// https://github.com/google/go-cloud/issues/2413 is fixed.
+	DocstoreRevision interface{}
+}
+
+// TODO(rvangent): This is pretty raw HTML. Should we have a common style sheet/etc. for demos?
```
Comment (clausti) [inline internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go]: common style sheet for demos sounds like a good idea to me!

File: internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+	if err := docstoreBaseTmpl.Execute(w, input); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// docstoreListHandler lists the items in a collection, possibly under a "prefix"
+// query parameter. Each listed directory is a link to list that directory,
+// and each non-directory is a link to view that file.
+func docstoreListHandler(w http.ResponseWriter, req *http.Request) {
+	input := &docstoreListData{URL: collectionURL}
```
Comment (jba) [inline internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go]: nit: it's odd that you call this `input` when it's the output.

Comment (vangent) [reply internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go]: I guess it depends on how you look at it; it's the input to the template, but the output of this function. I'll rename in a separate PR.

File: internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go (hunk 2 above)
Comment (vangent) [reply internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go]: Ack. I'll file an issue.

**Pull request merged**